src/index.ts

```diff
@@ -225,6 +225,14 @@ export class PartitionConsumer {
         await this.inner.stream(offset, event.emit.bind(event))
         return
     }
+
+    async getIterator(offset: Offset): Promise<any> {
+        let stream = await this.inner.getIterator(offset)
+        stream[Symbol.asyncIterator] = () => {
+            return stream
+        }
```

Comment:
@nicholastmosher What do you think of this cool line?

Reply:
I was actually looking at that... what the heck is happening here?

Reply:
Yeah... so an [`iterable` in javascript](https://javascript.info/iterable) needs to have a method `Symbol.iterator` and because this is a symbol, you can't just name the method in the node-bindgen macro(`name = "Symbol.iterator"`). I think that would be a `stream["Symbol.asyncIterator"] = () => { }` but it's unclear.

Anyway the `Symbol.iterator` function returns an object that has a `next()` method which returns an object with a  `done: bool` and a `value: Option<T>`. I personally find it annoying so, `getIterator` in the rust side of this returns an iterator which already conforms to the `.next()` convention already and so here it just returns itself.